modules/python3/processors/pythonscriptsource.cpp

```diff
@@ -0,0 +1,86 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/python3/processors/pythonscriptsource.h>
+#include <modules/python3/python3module.h>
+#include <inviwo/core/common/inviwoapplication.h>
+#include <inviwo/core/datastructures/geometry/basicmesh.h>
+#include <pybind11/pybind11.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo PythonScriptSource::processorInfo_{
+    "org.inviwo.PythonScriptSource",  // Class identifier
+    "Python Script Source",           // Display name
+    "Python",                         // Category
+    CodeState::Experimental,          // Code state
+    "Python, CPU",                    // Tags
+};
+const ProcessorInfo PythonScriptSource::getProcessorInfo() const { return processorInfo_; }
+
+PythonScriptSource::PythonScriptSource()
+    : Processor()
+    , script_("")
+    , scriptFileName_("scriptFileName", "File Name", "")
+    , mesh_("mesh")
+    , volume_("volume") {
+    addPort(mesh_);
+    addPort(volume_);
+
+    addProperty(scriptFileName_);
+
+    scriptFileName_.onChange([this]() { script_.setFilename(scriptFileName_.get()); });
+
+    script_.onChange([this]() { invalidate(InvalidationLevel::InvalidOutput); });
+}
+
+void PythonScriptSource::process() {
+    if (script_.getSource().empty()) {
+        mesh_.setData(std::make_shared<Mesh>());
+        return;
+    }
+    script_.run({}, [&](pybind11::dict dict) {
+        if (dict.contains("mesh")) {
+            auto pyMesh = dict["mesh"];
+            auto mesh = std::shared_ptr<BasicMesh>(pyMesh.cast<BasicMesh*>());
```

Comment:
BasicMesh->Mesh?

Reply:
inviwopy provides both `Mesh` and `BasicMesh`. But only `BasicMesh` provides the functionality for adding vertices like this (at least at the moment until the decorated mesh PR is approved and integrated into inviwopy):
```python
mesh = BasicMesh()
for p in pos :
    mesh.addVertex(vec3(p[0], p[1], p[2]), vec3(0), vec3(0), vec4(1, 0, 0, 1)
```